solr/solrj/src/java/org/apache/solr/client/solrj/impl/HttpSolrJdkClient.java

```diff
@@ -268,13 +258,14 @@ private HttpResponse<InputStream> doPutOrPost(
       bodyPublisher = HttpRequest.BodyPublishers.noBody();
     }
 
+    decorateRequest(reqb, solrRequest);
     if (method == SolrRequest.METHOD.PUT) {
-      reqb.PUT(bodyPublisher);
+      reqb.method("PUT", bodyPublisher);
     } else {
-      reqb.POST(bodyPublisher);
+      reqb.method("POST", bodyPublisher);
```

Comment:
could use one line (no if-else) via method.toString to get the string of the method